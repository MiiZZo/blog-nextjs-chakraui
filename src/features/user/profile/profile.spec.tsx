import React from "react";
import { cleanup } from "@testing-library/react";
import { myRender } from "@lib/testing/render";
import Profile from "./profile";

afterEach(cleanup);

describe("profile component", () => {
  const articles = [null, null, null];
  const user = {
    username: "username",
    bio: "my bio",
    homepage: "home.com",
    facebook: "facebook",
    twitter: "twitter",
    github: "github",
    registrationDate: "2020-08-06 23:55:06"
  };

  it("should match a shapshot", () => {
    const { asFragment } = myRender(
      <Profile user={user} articles={articles} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
