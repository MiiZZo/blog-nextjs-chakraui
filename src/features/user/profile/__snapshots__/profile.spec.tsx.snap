// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile component should match a shapshot 1`] = `
<DocumentFragment>
  <div
    class="css-o44is"
  >
    <div
      class="css-92qjs2"
    >
      <div
        class="css-0"
      >
        <div
          class="css-18o33ji"
        >
          <h2
            class="css-p5x3r9"
          >
            username
          </h2>
          <h2
            class="css-1yu9gkl"
          >
            Publications 3
          </h2>
        </div>
        <hr
          class="css-1fic0ux"
        />
        <h2
          class="css-100fc8h"
        >
          Bio
        </h2>
        <p
          class="css-bi78j0"
        >
          my bio
        </p>
        <h2
          class="css-bo2j60"
        >
          Links
        </h2>
        <ul
          class="css-1kjs0ga"
        >
          <li
            class="css-qp0z52"
          >
            <p
              class="css-1gl7g4v"
            >
              <span
                aria-label="home"
                class="anticon anticon-home css-1c8g4qf"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="home"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M946.5 505L534.6 93.4a31.93 31.93 0 00-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
                  />
                </svg>
              </span>
              <a
                class="css-u5zpo1"
                href="home.com"
              >
                home.com
              </a>
            </p>
          </li>
          <li
            class="css-qp0z52"
          >
            <p
              class="css-1gl7g4v"
            >
              <span
                aria-label="facebook"
                class="anticon anticon-facebook css-1c8g4qf"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="facebook"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                  />
                </svg>
              </span>
              <a
                class="css-u5zpo1"
                href="facebook"
              >
                facebook
              </a>
            </p>
          </li>
          <li
            class="css-qp0z52"
          >
            <p
              class="css-1gl7g4v"
            >
              <span
                aria-label="twitter-circle"
                class="anticon anticon-twitter-circle css-1c8g4qf"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="twitter-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm215.3 337.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 01-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 01-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 00229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
                  />
                </svg>
              </span>
              <a
                class="css-u5zpo1"
                href="twitter"
              >
                twitter
              </a>
            </p>
          </li>
          <li
            class="css-992tcj"
          >
            <p
              class="css-1gl7g4v"
            >
              <span
                aria-label="github"
                class="anticon anticon-github css-1c8g4qf"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="github"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                  />
                </svg>
              </span>
              <a
                class="css-u5zpo1"
                href="github"
              >
                github
              </a>
            </p>
          </li>
        </ul>
        <h2
          class="css-tr8dze"
        >
          Registration date
        </h2>
        <p
          class="css-fzcsno"
        >
          2020-08-06 23:55:06
        </p>
      </div>
    </div>
    <footer
      class="css-70aikg"
    >
      <ul
        class="css-19ndbdu"
      >
        <li
          class="css-992tcj"
        >
          <a
            class="css-u5zpo1"
          >
            Articles
          </a>
        </li>
        <li
          class="css-992tcj"
        >
          <a
            class="css-u5zpo1"
          >
            About
          </a>
        </li>
        <li
          class="css-992tcj"
        >
          <a
            class="css-u5zpo1"
          >
            Help
          </a>
        </li>
      </ul>
    </footer>
  </div>
</DocumentFragment>
`;
